pymatgen/analysis/chemenv/coordination_environments/chemenv_strategies.py

```diff
@@ -30,6 +30,9 @@
 from pymatgen.core.sites import PeriodicSite
 from pymatgen.symmetry.analyzer import SpacegroupAnalyzer
 
+if TYPE_CHECKING:
+    from typing_extensions import Self
+
```

Comment:
sneaky :)

`typing_extensions` is actually a [PyPI package](https://pypi.org/project/typing-extensions/) but given we never import it at runtime, looks like we can actually get away with not listing it as dependency and installing it in CI.
still, i think that might be bad practice as contributors who want to run type checks and don't have it installed will get errors (i assume). so maybe better to add to optional deps

Reply:
Good catch. I'm not even aware it's not a built-in package. We should be able to native `Self` support after Python3.11 but it looks we still need a legacy `typing_extensions` for now. Or I could still use the name of the Class itself instead of `Self`(but apparently `Self` is more proper), if you decide adding an extra optional dep is overkill for this.

After fixing it this PR should be good. Thanks!

Reply:
i'm a fan of `Self` and i don't mind depending on `typing_extensions` until we reach oldest supported Python = 3.11.

but maybe it would be better to add it in a separate PR and then use `Self` on _all_ methods where it applies?

Reply:
I help do this but I guess reverting changes in this PR would pointless then? Maybe add an optional dep here and I would do the following later?

Reply:
yes, that works

Reply:
Thanks. I would do that soon.

Reply:
(I'm hoping I'm not adding it to the wrong place)

Reply:
that's correct. i'm still not sure what happens to people who use an IDE with type hint support that don't have `typing_extensions` installed. could be that their IDE doesn't understand `Self` but would have understood the `NameOfClass` hint. could be we're removing some type assistance in which case to fix that we would even need to make it a real dependency. but this is all just me guessing. and either way, it doesn't seem like a big loss

Reply:
I'm working with VS Code and I tried to uninstall `typing_extensions` and everything seems to work just fine:
```bash
(venv) yang@Yang-PC:~/pymatgen$ pip show typing_extensions
WARNING: Package(s) not found: typing_extensions
(venv) yang@Yang-PC:~/pymatgen$ pre-commit run --all-files
ruff.....................................................................Passed
ruff-format..............................................................Passed
check yaml...............................................................Passed
fix end of files.........................................................Passed
trim trailing whitespace.................................................Passed
mypy.....................................................................Passed
codespell................................................................Passed
cython-lint..............................................................Passed
double-quote Cython strings..............................................Passed
blacken-docs.............................................................Passed
markdownlint.............................................................Passed
nbstripout...............................................................Passed
```

Meanwhile I'm pretty sure I didn't manually install `typing_extensions`, and it seems bundled with `venv` for some reason.